src/main/navigation/navigation.c

```diff
@@ -2792,6 +2796,18 @@ void getWaypoint(uint8_t wpNumber, navWaypoint_t * wpData)
         wpData->lon = wpLLH.lon;
         wpData->alt = wpLLH.alt;
     }
+    // WP #254 - special waypoint - get desiredPosition that was set by ground control station if in GCS assisted mode
+    else if (wpNumber == 254) {
+        if ((posControl.gpsOrigin.valid) && (posControl.flags.isGCSAssistedNavigationEnabled) && (posControl.navState == NAV_STATE_POSHOLD_3D_IN_PROGRESS)) {
```

Comment:
I think checks for GCS and 3D-poshold unnecessary limit the usefullness of this. In any 3D-guided mode (PH, RTH, WP, CRUISE etc) it might be useful to access the **current** target position, not necessarily for robotic applications. I think we can just check for `NAV_CTL_ALT | NAV_CTL_POS` bits in the current state flags (returned by `navGetStateFlags`)

Reply:
Replaced with check for 3D-guided mode (POS and ALT flags)